### PR TITLE
Emit events for host library actions & misc fixes

### PR DESCRIPTION
**Pull request by vishnumad**
### _Summary_
* Apply prettier config to match Coinbase style
* Add polyfills for `URL` and `Buffer` which aren't in React Native by default
* Add native function on Android for getting the intent URL `getAndroidIntentUrl`
* Add event emitter for host library actions

### CODE REVIEW

File: react-native/host/package.json [hunk 1]
```diff
@@ -91,10 +91,14 @@
...
-          "quoteProps": "consistent",
+          "arrowParens": "always",
+          "bracketSameLine": false,
+          "jsxSingleQuote": false,
+          "printWidth": 100,
+          "semi": true,
           "singleQuote": true,
           "tabWidth": 2,
-          "trailingComma": "es5",
+          "trailingComma": "all",
```
Comment (vishnumad) [inline react-native/host/package.json:101]: Matches prettier style used by CB internally

File: react-native/host/package.json [hunk 2]
```diff
@@ -124,5 +132,9 @@
         }
       ]
     ]
+  },
+  "dependencies": {
+    "buffer": "^6.0.3",
+    "react-native-url-polyfill": "^1.3.0"
```
Comment (vishnumad) [inline react-native/host/package.json:138]: Polyfills URL and Buffer which don't have complete implementations in React Native

File: react-native/host/src/events/events.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+  | SessionNotFoundEvent
+  | SessionExpiredEvent
+  | SessionAddedEvent
+  | EncodeResponseStartEvent
+  | EncodeResponseSuccessEvent
+  | EncodeResponseFailureEvent
+  | SendSuccessResponseEvent
+  | SendFailureResponseEvent;
+
+const events = new EventEmitter();
```
Comment (bangtoven) [inline react-native/host/src/events/events.ts]: should we use this name as well?
https://github.com/coinbase/coinbase-wallet-sdk/blob/master/packages/wallet-sdk/src/connection/DiagnosticLogger.ts

**Review by bangtoven — COMMENTED** [review]
@vishnumad

**Comment by AMITGOELNY** [comment]
Looks mostly good.  I would normally advise splitting up any formatting into separate PR, but this is fine for the PR size.

File: react-native/host/src/events/events.ts (hunk 3 above)
Comment (vishnumad) [reply react-native/host/src/events/events.ts]: Renamed [9b0a6af](https://github.com/coinbase/wallet-mobile-sdk/pull/130/commits/9b0a6af180efbf91858faac3d8f19dcce7de824e)

**Review by AMITGOELNY — APPROVED** [review]

**Pull request merged**